htdocs/core/class/extrafields.class.php

```diff
@@ -1254,7 +1254,7 @@ public function showInputField($key, $value, $moreparam = '', $keysuffix = '', $
 					dol_syslog(get_class($this).'::showInputField type=sellist', LOG_DEBUG);
 					$resql = $this->db->query($sql);
 					if ($resql) {
-						$out .= '<option value="0">&nbsp;</option>';
+						if (!$required) $out .= '<option value="0">&nbsp;</option>';
```

Comment:
If tou don't add the zmpty value when field is required, it means the system will auto select the first value in list. This will save one click, avoinding user to select a value, but will being a lot of problem of data not correctly set or corrupted because value was set randomly (first found) instead of explicitely selected by end user.
This is why we must keep the empty value. The mandatory status will be blocked later if user does not set explicitely a value.